cratetorrent/src/tracker.rs

```diff
@@ -140,18 +152,72 @@ pub(crate) struct Tracker {
     client: Client,
     /// The URL of the tracker.
     url: Url,
+    protocol: NetProtocol,
 }
 
 impl Tracker {
-    pub fn new(url: Url) -> Self {
+    pub fn new(url: TrackerUrl) -> Self {
         Self {
             client: Client::new(),
-            url,
+            url: url.url,
+            protocol: url.protocol,
         }
     }
 
+    ///https://www.bittorrent.org/beps/bep_0015.html
+    async fn connect_udp(ip_addr: SocketAddr) -> Option<i64> {
+        //Bind to a random port
+        let port = thread_rng().gen_range(1025..u16::MAX);
+
+        let mut sock = UdpSocket::bind(SocketAddr::from(([0, 0, 0, 0], port)))
+            .await
+            .unwrap();
+
+        //The magic protocol id number
```

Comment:
Could you please leave a white space between the last forward slash and the first character of the comment? Same goes for the other places.

Reply:
Will do!